Source/WebKit/WebProcess/cocoa/WebProcessCocoa.mm

```diff
@@ -662,18 +662,36 @@ static void registerLogHook()
     os_log_set_hook(OS_LOG_TYPE_DEFAULT, ^(os_log_type_t type, os_log_message_t msg) {
         if (msg->buffer_sz > 1024)
             return;
-        char* messageString = os_log_copy_message_string(msg);
-        String logString = String::fromUTF8(messageString);
-        free(messageString);
 
+        String logFormat = String::fromUTF8(msg->format);
         String logChannel = String::fromUTF8(msg->subsystem);
         String logCategory = String::fromUTF8(msg->category);
 
-        callOnMainRunLoop([logChannel = logChannel.isolatedCopy(), logCategory = logCategory.isolatedCopy(), logString = logString.isolatedCopy(), type] {
-            auto* connection = WebProcess::singleton().existingNetworkProcessConnection();
-            if (!connection)
-                return;
-            connection->connection().send(Messages::NetworkConnectionToWebProcess::LogOnBehalfOfWebContent(logChannel, logCategory, logString, type, getpid()), 0);
+        Vector<uint8_t> buffer;
+        buffer.resize(msg->buffer_sz);
```

Comment:
No need to call resize(), you can pass a parameter to the Vector constructor to give it the appropriate size right away. Pretty sure there is even a constructor where you can pass the data you want to memcpy right away.